embulk-core/src/main/java/org/embulk/spi/ExecSession.java

```diff
@@ -196,4 +222,62 @@ public boolean isPreview() {
     public void cleanup() {
         tempFileSpace.cleanup();
     }
+
+    private static Optional<Instant> toInstantFromString(final String string) {
```

Comment:
I'd be more confident having some way to test these private methods, but I leave it to you.

Reply:
Agreed. Added tests in fe530f0ffd14e706df86157c80d14ad056ac700a

Reply:
Thanks!